src/network/networkd-dhcp-server.c

```diff
@@ -267,8 +355,10 @@ int dhcp4_server_configure(Link *link) {
                                                        "Not emitting %s on link, couldn't find suitable uplink.",
                                                        dhcp_lease_info_to_string(n));
                                         r = 0;
-                                } else
+                                } else if (uplink->network)
                                         r = link_push_uplink_to_dhcp_server(uplink, n, link->dhcp_server);
+                                else
```

Comment:
hmm, this lacks a check `n == SD_DHCP_LEASE_DNS_SERVERS` actually, otherwise you push DNS server info into NTP and so on...

Reply:
right!